packages/thirdweb/src/extensions/farcaster/signers.ts

```diff
@@ -0,0 +1,25 @@
+import { fromBytes } from "../../utils/encoding/from-bytes.js";
+import type { Hex } from "../../utils/encoding/hex.js";
+
+export type Ed25519Keypair = {
+  publicKey: Hex;
+  privateKey: Hex;
+};
+
+/**
+ * Generates an Ed25519 keypair to be used as an account signer.
+ * @returns A promise resolving to the generated keypair.
+ * @example
+ * ```
+ * createSigner()
+ * ```
+ */
+export async function createSigner(): Promise<Ed25519Keypair> {
```

Comment:
probably makes sense to move this out of this extension and into the `/wallets` namespace - this is basically the beginning of a function we want to have anyways:
- `createRandomAccount()` which is supposed to do this same logic and then return a valid `Account` at the end

Reply:
This signer actually uses EdDSA rather than ECDSA, so it's not an ethereum-compatible account. Would you still want it added to `/wallets`. Either way I could rename it to `createEdDSAKey` or `createFarcasterKey` to reduce confusion?

Side note, I can add the `createRandomAccount` function in a new PR

Reply:
okay in this case it can probably stay in farcaster extension (until we have another use-case for this specific curve down the line)

and yes RE the createRandomAccount, that would be great!